src/subprocess-posix.cc

```diff
@@ -117,20 +120,42 @@ bool Subprocess::Start(SubprocessSet* set, const string& command) {
   if (err != 0)
     Fatal("posix_spawnattr_setflags: %s", strerror(err));
 
+  if (use_console_) {
+    // Block SIGCHLD to prevent a race where the new subprocess exits before we
+    // register its PID.
```

Comment:
This should not be necessary as SIGCHLD should be blocked when this function is called. See comment below for more details.

Apart from that, a proper version of this code should save the old procmask and restore it on line 146 instead.